src/Microsoft.AspNetCore.Mvc.Core/ControllerBase.cs

```diff
@@ -1482,5 +1486,12 @@ public virtual SignOutResult SignOut(AuthenticationProperties properties, params
                 model: model);
             return ModelState.IsValid;
         }
+
+        private Task<CompositeValueProvider> GetValueProviderAsync()
```

Comment:
The other pivot here that I considered is to add `ICompositeValueProviderFactory` as a public property on `ControllerBase`. This would improve the testability and help make these overloads of `TryUpdateModel` that take a value provider more useful.

However it would add yet another thing to the controller surface area...